src/lib/components/UrlChooser/index.js

```diff
@@ -0,0 +1,185 @@
+import {html} from "lit-element";
+import {store} from "../../store";
+import {BaseElement} from "../BaseElement";
+import {userRef, firebase} from "../../fb";
+
+/* eslint-disable require-jsdoc */
+class UrlChooser extends BaseElement {
+  static get properties() {
+    return {
+      href: {type: String},
+      switching: {type: Boolean, reflect: true},
+      active: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.href = null; // when signed out or waiting for Firestore, this is null
+    this.switching = true;
+    this.active = false;
+
+    // non-properties
+    this.urlInput = null;
+    this.runLighthouseButton = null;
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+  }
+
+  render() {
+    return html`
+      <div class="report_header_enterurl">
+        <div class="lh-enterurl lh-enterurl--selected">${this.href}</div>
+        <div class="lh-enterurl lh-enterurl--switch">
+          <input
+            type="url"
+            class="lh-input"
+            .value="${this.href}"
+            placeholder="Enter a web page URL"
+            pattern="https?://.*"
+            minlength="7"
+            required
+            @keyup="${this.onUrlKeyup}"
+          />
+          <button
+            class="lh-enterurl__close"
+            aria-label="Remove URL"
+            @click=${this.clearInput}
+          ></button>
+        </div>
+        <div class="lh-controls">
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--secondary"
+            @click=${this.switchUrl}
+          >
+            Switch URL
+          </button>
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--primary"
+            id="run-lh-button"
+            @click=${this.runAudit}
+          >
+            Run Audit
+          </button>
+        </div>
+      </div>
+    `;
+  }
+
+  updated() {
+    this.urlInput = this.renderRoot.querySelector('input[type="url"]');
+    this.runLighthouseButton = this.renderRoot.querySelector("#run-lh-button");
+  }
+
+  onStateChanged() {
+    const state = store.getState();
+    const urlInputValue = this.urlInput ? this.urlInput.value : null;
+
+    if (this.active) {
```

Comment:
This seems like something that would be better tracked as state in the store (since other things may depend on it) versus the component itself. Although I do realize that the component might change its visual state when LH is running. Perhaps that info should be passed down to the component. See my comments below about Containers and state.

Reply:
I've broken them apart.